custom_components/tahoma/cover.py

```diff
@@ -102,49 +102,26 @@ class TahomaCover(TahomaDevice, CoverEntity):
     def current_cover_position(self):
         """Return current position of cover."""
 
-        states = self.tahoma_device.active_states
-
-        position = None
-
-        # Set position for vertical covers
-        if CORE_CLOSURE_STATE in states:
-            position = 100 - states.get(CORE_CLOSURE_STATE)
-
-        # Set position for horizontal covers
-        if CORE_DEPLOYMENT_STATE in states:
-            position = 100 - states.get(CORE_DEPLOYMENT_STATE)
-
-        # Set position for gates
-        if CORE_PEDESTRIAN_POSITION_STATE in states:
-            position = 100 - states.get(CORE_PEDESTRIAN_POSITION_STATE)
-
-        if CORE_TARGET_CLOSURE_STATE in states:
-            position = 100 - states.get(CORE_TARGET_CLOSURE_STATE)
-
-        if position is not None:
-            # HorizontalAwning devices need a reversed position that can not be obtained via the API
-            if "Horizontal" in self.tahoma_device.widget:
-                position = 100 - position
-
-            # TODO Check if this offset is really necessary
-            if position <= 5:
-                position = 0
-            if position >= 95:
-                position = 100
+        position = self.select_state(
+            CORE_CLOSURE_STATE,
+            CORE_DEPLOYMENT_STATE,
+            CORE_PEDESTRIAN_POSITION_STATE,
+            CORE_TARGET_CLOSURE_STATE,
+        )
 
-        return position
+        if "Horizontal" in self.tahoma_device.widget:
+            return position
+        else:
+            return 100 - position
```

Comment:
I'm surprise you didn't use the ternary operator here :D Usually you like it.

Reply:
lol I know... I don't understand why they don't have `test ? opt1 : opt2` in python, so much better

Reply:
I can change that if you'd like ;-)